initrd/bin/oem-factory-reset

```diff
@@ -1211,139 +1299,155 @@ PUBKEY="/tmp/${GPG_GEN_KEY}.asc"
+	if [ -e /etc/config.user ]; then
+		cbfs.sh -o /tmp/oem-setup.rom -a "heads/initrd/etc/config.user" -f /etc/config.user
+	fi
+
+	# flash updated firmware image
+	echo -e "\nAdding generated key to current firmware and re-flashing...\n"
+	if ! /bin/flash.sh /tmp/oem-setup.rom 2>/tmp/error; then
+		ERROR=$(tail -n 1 /tmp/error | fold -s)
+		whiptail_error_die "Error flashing updated firmware image:\n\n$ERROR"
+	fi
 fi
 
 ## sign files in /boot and generate checksums
 if [[ "$SKIP_BOOT" == "n" ]]; then
-    echo -e "\nSigning boot files and generating checksums...\n"
-    generate_checksums
+	echo -e "\nUpdating checksums and signing all files in /boot...\n"
+	generate_checksums
 fi
 
 # passphrases set to be empty first
-passphrases="\n"
+passphrases=""
 
 # Prepare whiptail output of configured secrets
 if [ -n "$luks_new_Disk_Recovery_Key_passphrase" -o -n "$luks_new_Disk_Recovery_Key_passphrase_desired" ]; then
-    passphrases+="LUKS Disk Recovery Key passphrase: ${luks_new_Disk_Recovery_Key_passphrase}\n"
+	passphrases+="LUKS Disk Recovery Key passphrase: ${luks_new_Disk_Recovery_Key_passphrase}\n"
 fi
 
 if [ "$CONFIG_TPM" = "y" ]; then
-    passphrases+="TPM Owner Password: ${TPM_PASS}\n"
+	passphrases+="TPM Owner Password: ${TPM_PASS}\n"
+fi
+
+#if nk3 detected, we add the NK3 Secre App PIN. Detect by product ID
+if lsusb | grep -q "20a0:42b2"; then
+	passphrases+="Nitrokey 3 Secrets app PIN: ${ADMIN_PIN}\n"
 fi
 
 #GPG PINs output
 passphrases+="GPG Admin PIN: ${ADMIN_PIN}\n"
 #USER PIN was configured if GPG_GEN_KEY_IN_MEMORY is not active or if GPG_GEN_KEY_IN_MEMORY_COPY_TO_SMARTCARD is active
 if [ "$GPG_GEN_KEY_IN_MEMORY" = "n" -o "$GPG_GEN_KEY_IN_MEMORY_COPY_TO_SMARTCARD" = "y" ]; then
-    passphrases+="GPG User PIN: ${USER_PIN}\n"
+	passphrases+="GPG User PIN: ${USER_PIN}\n"
 fi
 
 #If user decided to generate keys in memory, we add the thumb drive passphrase
 if [ "$GPG_GEN_KEY_IN_MEMORY" = "y" ]; then
-    passphrases+="GPG key material backup passphrase: ${ADMIN_PIN}\n"
+	passphrases+="GPG key material backup passphrase: ${ADMIN_PIN}\n"
 fi
 
-## Show to user current configured secrets prior of rebooting
-whiptail --msgbox "
-    $(echo -e "$passphrases" | fold -w $((WIDTH-5)))" \
-    $HEIGHT $WIDTH --title "Configured secrets"
+# Show configured secrets in whiptail and loop until user confirms qr code was scanned
+while true; do
+	whiptail --msgbox "$(echo -e "$passphrases" | fold -w $((WIDTH - 5)))" \
+		$HEIGHT $WIDTH --title "Configured secrets"
+	#Tell user to scan the QR code containing all configured secrets
+	echo -e "\nScan the QR code below to save the secrets to a secure location"
+	qrenc "$(echo -e "$passphrases")"
+	# Prompt user to confirm scanning of qrcode on console prompt not whiptail: y/n
+	echo -e -n "Please confirm you have scanned the QR code above and/or written down the secrets? [y/N]: "
+	read -n 1 prompt_output
+	echo
+	if [ "$prompt_output" == "y" -o "$prompt_output" == "Y" ]; then
+		break
+	fi
```

Comment:
Why do we need to loop here rather than our usual "press enter"?  If somebody forgets to scan the QR code, you can always OEM reset again

Reply:
@JonathonHall-Purism to make sure they understand that qr code here contains all providioned secrets and have to type Y (forced input)